From 28c1c30ea8676f7939c6d4c90aac714bb11264c6 Mon Sep 17 00:00:00 2001
From: Shuo Qian <shuoq@google.com>
Date: Fri, 5 Mar 2021 09:58:19 -0800
Subject: [PATCH] [Q_asb_2021-07] Add permission for the broadcast intent with
 SIP profiles for security purpose.

Test: Treehugger
Bug: 176496502
Change-Id: I2087170aceb7c0670afd04fcc779047aa95a373f
(cherry picked from commit 0e459673aa944d65989181b659c820504117ab51)
---
 src/java/com/android/server/sip/SipService.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/java/com/android/server/sip/SipService.java b/src/java/com/android/server/sip/SipService.java
index d14a8d6..5328801 100644
--- a/src/java/com/android/server/sip/SipService.java
+++ b/src/java/com/android/server/sip/SipService.java
@@ -342,7 +342,7 @@ public final class SipService extends ISipService.Stub {
         if (DBG) log("notify: profile added: " + localProfile);
         Intent intent = new Intent(SipManager.ACTION_SIP_ADD_PHONE);
         intent.putExtra(SipManager.EXTRA_LOCAL_URI, localProfile.getUriString());
-        mContext.sendBroadcast(intent);
+        mContext.sendBroadcast(intent, android.Manifest.permission.USE_SIP);
         if (mSipGroups.size() == 1) {
             registerReceivers();
         }
@@ -352,7 +352,7 @@ public final class SipService extends ISipService.Stub {
         if (DBG) log("notify: profile removed: " + localProfile);
         Intent intent = new Intent(SipManager.ACTION_SIP_REMOVE_PHONE);
         intent.putExtra(SipManager.EXTRA_LOCAL_URI, localProfile.getUriString());
-        mContext.sendBroadcast(intent);
+        mContext.sendBroadcast(intent, android.Manifest.permission.USE_SIP);
         if (mSipGroups.size() == 0) {
             unregisterReceivers();
         }
-- 
2.34.1

